packages/apps/human-app/server/src/integrations/reputation-oracle/reputation-oracle.gateway.ts

```diff
@@ -0,0 +1,109 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { lastValueFrom } from 'rxjs';
+import { InjectMapper } from '@automapper/nestjs';
+import { Mapper } from '@automapper/core';
+import {
+  SignupWorkerCommand,
+  SignupWorkerData,
+} from '../../modules/user-worker/interfaces/worker-registration.interface';
+import {
+  SignupOperatorCommand,
+  SignupOperatorData,
+} from '../../modules/user-operator/interfaces/operator-registration.interface';
+import { GatewayConfigService } from '../../common/config/gateway-config.service';
+import { GatewayConfig } from '../../common/interfaces/endpoint.interface';
+import { ExternalApiName } from '../../common/enums/external-api-name';
+import { EndpointName } from '../../common/enums/endpoint-name';
+import { AxiosRequestConfig } from 'axios';
+import {
+  RequestDataType
+} from './reputation-oracle.interface';
+import {
+  SigninWorkerCommand,
+  SigninWorkerData, SigninWorkerResponse,
+} from '../../modules/user-worker/interfaces/worker-signin.interface';
+
+@Injectable()
+export class ReputationOracleGateway {
+  private readonly reputationOracleConfig: GatewayConfig;
+  constructor(
+    private httpService: HttpService,
+    gatewayConfigService: GatewayConfigService,
+    @InjectMapper() private readonly mapper: Mapper,
+  ) {
+    this.reputationOracleConfig = gatewayConfigService.getConfig(
+      ExternalApiName.REPUTATION_ORACLE,
+    );
+  }
+  private getEndpointOptions(
+    endpointName: EndpointName,
+    data: RequestDataType,
+  ) {
+    const { method, endpoint, headers } =
+      this.reputationOracleConfig.endpoints[endpointName];
+    return {
+      method: method,
+      url: `${this.reputationOracleConfig.url}${endpoint}`,
+      headers: headers,
+      data: data,
+    };
+  }
+
+  private async handleRequestToReputationOracle<T>(
+    options: AxiosRequestConfig,
+  ): Promise<T> {
+    try {
+      const response = await lastValueFrom(this.httpService.request(options));
+      return response.data;
+    } catch (error) {
+      if (error.response) {
+        throw new HttpException(error.response.data, error.response.status);
```

Comment:
@BartoszSolkaBD any particular reasons to throw `HttpException` here? I mean, technically, it's a service level which shouldn't be responsible for that.
Instead, we could throw here an app error and use nest exception filters: https://docs.nestjs.com/exception-filters.
Please, let me know what do you think about that.

Reply:
For now, we don't see benefits from the suggested refactoring. The current solution is more readable in our opinion. Additionally, it's similar to how the error handling is implemented in the current implementation of the Reputation Oracle server

Reply:
Tbh I wouldn't recommend using a reputation oracle server implementation as a reference because it's going to be refactored quite heavily since it's just not following a lot of conventions.
Well, you could leave it as it but please keep in mind that during the 2nd milestone we would ask you to change it.

Reply:
okay, fine for me